### ISSUE TITLE
Add dt.monthend() and dt.monthstart()

### ISSUE THREAD

**Issue by mcrumiller** [pola-rs/polars#8161]
### Problem description

Having `Expr.dt.monthend()` and `Expr.dt.monthstart()` would be very handy. They can currently be implemented with expressions but can probably have faster native implementations, and they are common enough operations that I think they should be added to the API.

```python
from polars import col

def monthbegin(col_name):
    return col(col_name) - pl.duration(days=col(col_name).dt.day()-1)

def monthend(col_name):
    return (col(col_name) - pl.duration(days=col(col_name).dt.day()-1)).offset_by("1mo").dt.offset_by("-1d")
```

**Comment by MarcoGorelli** [comment]
For month-begin, you can already just use `truncate`
```python
In [82]: df
Out[82]:
shape: (1, 1)
┌─────────────────────┐
│ ts                  │
│ ---                 │
│ datetime[μs]        │
╞═════════════════════╡
│ 2020-04-14 00:00:00 │
└─────────────────────┘

In [83]: df.with_columns(
    ...:     month_begin=pl.col('ts').dt.truncate('1mo'),
    ...:     month_end=pl.col('ts').dt.truncate('1mo').dt.offset_by('1mo').dt.offset_by('-1d')
    ...: )
Out[83]:
shape: (1, 3)
┌─────────────────────┬─────────────────────┬─────────────────────┐
│ ts                  ┆ month_begin         ┆ month_end           │
│ ---                 ┆ ---                 ┆ ---                 │
│ datetime[μs]        ┆ datetime[μs]        ┆ datetime[μs]        │
╞═════════════════════╪═════════════════════╪═════════════════════╡
│ 2020-04-14 00:00:00 ┆ 2020-04-01 00:00:00 ┆ 2020-04-30 00:00:00 │
└─────────────────────┴─────────────────────┴─────────────────────┘
```

month-end requires using `offset_by`, but still, it's quite simple (personally, I'd suggest not expanding the API to include them)

**Comment by mcrumiller** [comment]
@MarcoGorelli thanks, those are probably more performant than my expressions.

I still think they are common enough (and not obvious enough, as I didn't realize we could use `truncate`) to be added.

**Comment by stinodego** [comment]
Please name them `dt.month_end` and `dt.month_start` then - to be consistent with our naming conventions :)

**Comment by mcrumiller** [comment]
This should have been closed via #8469.
